h2quic/client.go

```diff
@@ -178,7 +178,7 @@ func (c *client) RoundTrip(req *http.Request) (*http.Response, error) {
 		requestedGzip = true
 	}
 	// TODO: add support for trailers
-	endStream := !hasBody
+	endStream := !hasBody && req.Method != http.MethodConnect
```

Comment:
According to the documentation (e.g. here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT) it looks like the request should have a body. The http2 package in the standard library doesn't do any checks for the request method here as well.
Can you explain why you think we need this addition here?